### ISSUE TITLE
Fix Node Tests

### ISSUE THREAD

**Issue by pbimpisidis** [dfensgmbh/biz.dfch.PS.Appclusive.Client#47]
- [x] Update file Node.Tests
- [x] Update CLOUDTCL-1873

**Comment by pbimpisidis** [comment]
@rufer7 in the file there are some tests that are not in the jira ticket CLOUDTCL-1873:
CreateWithJobConditionParametersSucceeds
DoStateChangeOnNodeSetsConditionAndConditionParametersOnJob
GetAssignablePermissionsForConfigurationNode-ReturnsIntrinsicEntityKindNonNodePermissions
GetAssignablePermissionsForRootNode-ReturnsPermissionsExceptIntrinsicEntityKindNonNodePermissions

the last 3 return errors. Do I keep them in the file? If yes can you explain to me the last 3 so I can also write the test case for jira?

**Comment by rufer7** [comment]
@pbimpisidis we won't remove test only because there is no issue in Jira. 

* CreateWithJobConditionParametersSucceeds
  There is a special action on the NodesController that allows creation of a node including passing the conditionParameters of the corresponding job. 

* DoStateChangeOnNodeSetsConditionAndConditionParametersOnJob
  The idea behind this test is that a node gets created and implicitly a job for this node gets created which is in the `InitialState` and the Condition is set to `Initialise`. This means the job state transition is not yet completed. To complete the state transition of the job, which brings the job into the next state, we simulate the worker by calling `JobResult` action, which indicates a successful state transition and the job reaches the next state. That's the setup for this test. Now in as the job is in a state wihout an active state transition we call the `InvokeAction` to initiate a state transition. `InvokeAction` sets the condition (must be valid) on the job and the jobConditionParameters according the parameters passed via `InvokeAction`

* GetAssignablePermissionsForConfigurationNode-ReturnsIntrinsicEntityKindNonNodePermissions
  This test tests that requesting permissions to assign to a configuration node (EntityKindId = 33) only returns permissions related to intrinsic entity kinds except node related permissions (i.e. Approval, Job, etc. are intrinsic entityKinds -> only permissions related to Approval, Job, etc. will be returned)

* GetAssignablePermissionsForRootNode-ReturnsPermissionsExceptIntrinsicEntityKindNonNodePermissions
  For root nodes (EntityKindId = 29) only node related and non intrinsic EntityKind related permissions get returned when querying assignable permssions.

To fix the last 3 tests try to do the following:

* Replace hardcoded ParentId with a variable that gets assigned with `(Get-ApcTenant -Current).NodeId`
* Replace hardcoded EntityKindId with an entityKindId gt 4096:
  
```
$query = "Id gt {0}" -f [biz.dfch.CS.Appclusive.Public.Constants+EntityKindId]::ReservationEnd.value__;
		$ekId = ($svc.Core.EntityKinds.AddQueryOption('$filter', $query) | Select -First 1).id;
```

Afterwards I expect the tests to succeed.

**Comment by pbimpisidis** [comment]
@rufer7 thanks a lot for the explanation, I will apply the changes

**Comment by rufer7** [comment]
Resolved by #74
